lib/ruby_lsp/document.rb

```diff
@@ -36,9 +43,12 @@ def initialize(source:, version:, uri:, encoding: Encoding::UTF_8)
 
     sig { params(other: Document).returns(T::Boolean) }
     def ==(other)
-      @source == other.source
+      self.class == other.class && uri == other.uri && @source == other.source
```

Comment:
I don't believe we need the class part since the URI is already a unique identifier for the document.
```suggestion
      uri == other.uri && @source == other.source
```

Reply:
Okay, that's fine.